### DISCUSSION TITLE
Alpine Expression Error: Cannot set properties of null (setting '_x_dataStack')

### DISCUSSION THREAD

**Discussion by C-Mejlak** [alpinejs/alpine#2393] [1. Help]
I'm getting this error, however properties are available in `x-data`. In local development it is working fine

![image](https://user-images.githubusercontent.com/39129161/142716298-5d4fad06-cebc-48f0-9862-f90f17751343.png)

![image](https://user-images.githubusercontent.com/39129161/142716322-4320cc87-7587-4350-8d5a-0d8fbc212d86.png)

Any pointers to why this is happening?

Thanks 🙏

**Comment by SimoTod** [comment]
You need to post your component otherwise it's impossible to debug. Are you initializing properties in x-init by any chance? Do you have a sensible default before it's ready (an empty array)?

**Comment by C-Mejlak** [comment]
Here's the component, yes I'm already initializing `properties` as an empty array. Really lost why this could be happening.

```html
<script>
  document.addEventListener("alpine:init", () => {
    Alpine.data("propertiesData", () => ({
      properties: [],
     
      ...

      async init() {
        const res = await fetch(`${location.origin}/json/properties.json`)
        const data = await res.json()
        this.properties = data.properties   
      }

    }))
  })
</script>
<div x-data="propertiesData">
  <template x-for="property in properties">...</template>
</div>
```

**Reply by C-Mejlak** [reply]
It was actually/accidentally working but not anymore, if it helps it's happening on https://kingly-helicopter.cloudvent.net/ once you hit 'Search"

**Reply by SimoTod** [reply]
I'm not at my laptop so i can't check. Try to console log data properties to make sure you don't get back anything unexpected

**Reply by C-Mejlak** [reply]
Everything looks fine, banging my head on this one

![image](https://user-images.githubusercontent.com/39129161/142720568-97d3f127-8f87-4735-b2d6-37377f664fbf.png)

**Reply by SimoTod** [reply]
That URL doesn't work for me, I've created a simple mock API and your code logic works okay for me: https://codepen.io/SimoTod/pen/BadbbWW.
I get the error if I try using a text node like the script above though `<template x-for="property in properties">...</template>` but I assume you don't use it like that right? It wouldn't make much sense.

**Reply by C-Mejlak** [reply]
Thanks for taking the time @SimoTod, yeah it's working fine on my dev environment. It must be something in the production environment messing something up.

Haven't had the time to look into it as of yet.
Thanks again!
